### ISSUE TITLE
Debugging using npm script stopped working

### ISSUE THREAD

**Issue by kgonyon** [microsoft/vscode-js-debug#577]
My team uses node scripts to start our nodejs projects and debug through VSCode. With the new VSCode update they have ceased to work anymore. I created a minimum setup below.
```
// .vscode/launch.json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug",
      "type": "node",
      "request": "launch",
      "cwd": "${workspaceRoot}",
      "runtimeExecutable": "npm",
      "runtimeArgs": [
          "run-script", "debug"
      ],
      "port": 5858
    }
  ]
}
```
```
// package.json
{
  "scripts": {
    "debug" : "node --inspect-brk=5858 index.js",
    "start" : "node index.js"
  }
}
```
```javascript
// index.js
console.log("Ran");
```

For some reason after the most recent update the debugger hangs when you run Debug. It prints the following.
```bash
/Users/kgonyon/.nvm/versions/node/v12.18.0/bin/npm run-script debug
Debugger listening on ws://127.0.0.1:51138/d4db49f0-bcae-49f5-98dc-8accbf7198d7
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.

> @ debug /Users/kgonyon/Projects/test
> node --inspect-brk=5858 index.js
```
If I run the debug script in the terminal and then attach via VSCode it works fine.

**To Reproduce**
1. Create Project with above files.
2. In VSCode run Debug

**Log File**
[vscode-debugadapter-1.json.gz](https://github.com/microsoft/vscode-js-debug/files/4905698/vscode-debugadapter-1.json.gz)

**VS Code Version:** 1.47.0
**Node Version:** 12.18.0

**Comment by connor4312** [comment]
Thanks for the issue! This is the same as https://github.com/microsoft/vscode/issues/99526#issuecomment-640771550

You can get around it by adjusting your launch config to:

```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug",
      "type": "node",
      "request": "launch",
      "program": "${workspaceFolder}/index.js",
      "stopOnEntry": true
    }
  ]
}
```

**Comment by connor4312** [comment]
Hi, following up on this issue after feedback from you and other users. There are two things to know:

1. If you don't need to debug on the first line of your program, it's actually safe to just remove `--inspect-brk` from your npm scripts with the new debugger. Previously, it was useful to prevent your program from running before VS Code could attach to it, but this is no longer necessary.

1. We've added some change to gracefully fall back (with a warning) when running a program with --inspect-brk in the config. This flag prevents us from being able to attach to children and unlock some additional new functionality, but it should work. This change will be available in the recovery release coming next week, and is available on our [nightly extension](https://github.com/microsoft/vscode-js-debug/#nightly-extension).

Happy Coding!

**Comment by kgonyon** [comment]
Thank you for the quick replies! With the information from your reply and the other ticket we were able to get a setup working we are happy with on the new debugger. Our last setup was just something that got created in a rush and copied to other projects. We found some other cool things along the way like recommending extensions for a project and auto starting typescript watch. So this turned out to be a blessing in disguise. Thank you again and to the team for making great tools.
